Client/Frontend/Brave Rewards/Panel/BraveRewardsViewController.swift

```diff
@@ -0,0 +1,207 @@
+        rewardsView.subtitleLabel.text = rewards.isEnabled ? Strings.Rewards.enabledBody : Strings.Rewards.disabledBody
+        rewardsView.rewardsToggle.addTarget(self, action: #selector(rewardsToggleValueChanged), for: .valueChanged)
+        rewardsView.legacyWalletTransferButton.addTarget(self, action: #selector(tappedRewardsTransfer), for: .touchUpInside)
+        rewardsView.legacyWalletTransferButton.dismissButton.addTarget(self, action: #selector(tappedDismissRewardsTransfer), for: .touchUpInside)
+        
+        view.snp.makeConstraints {
+            $0.width.equalTo(360)
+            $0.height.equalTo(rewardsView)
+        }
+        
+        if !AppConstants.buildChannel.isPublic {
+            let tapGesture = UITapGestureRecognizer(target: self, action: #selector(tappedHostLabel(_:)))
+            rewardsView.publisherView.hostLabel.isUserInteractionEnabled = true
+            rewardsView.publisherView.hostLabel.addGestureRecognizer(tapGesture)
+        }
+    }
+    
+    // MARK: - Actions
+    
+    private var isCreatingWallet: Bool = false
+    @objc private func rewardsToggleValueChanged() {
+        let isOn = rewardsView.rewardsToggle.isOn
+        rewards.isEnabled = isOn
+        rewardsView.subtitleLabel.text = isOn ? Strings.Rewards.enabledBody : Strings.Rewards.disabledBody
+        if rewardsView.rewardsToggle.isOn {
+            rewardsView.statusView.setVisibleStatus(status: supportedListCount > 0 ? .rewardsOn : .rewardsOnNoCount)
+        } else {
+            rewardsView.statusView.setVisibleStatus(status: .rewardsOff)
+        }
+        if publisher != nil {
+            UIView.animate(withDuration: 0.15) {
+                self.rewardsView.publisherView.isHidden = !self.rewardsView.rewardsToggle.isOn
+                self.rewardsView.publisherView.alpha = self.rewardsView.rewardsToggle.isOn ? 1.0 : 0.0
+            }
+        }
+    }
+    
+    @objc private func tappedRewardsTransfer() {
+        actionHandler?(.rewardsTransferTapped)
+    }
+    
+    @objc private func tappedDismissRewardsTransfer() {
+        Preferences.Rewards.dismissedLegacyWalletTransfer.value = true
+        UIView.animate(withDuration: 0.15) {
+            self.rewardsView.legacyWalletTransferButton.isHidden = true
+            self.rewardsView.legacyWalletTransferButton.alpha = 0.0
+        }
+    }
+    
+    @objc private func tappedUnverifiedPubLearnMore() {
+        actionHandler?(.unverifiedPublisherLearnMoreTapped)
+    }
+    
+    // MARK: - Debug Actions
+    
+    @objc private func tappedHostLabel(_ gesture: UITapGestureRecognizer) {
+        if gesture.state != .ended { return }
+        guard let publisher = publisher else { return }
+        rewards.ledger.refreshPublisher(withId: publisher.id) { [weak self] status in
+            guard let self = self else { return }
+            let copy = publisher.copy() as! PublisherInfo // swiftlint:disable:this force_cast
```

Comment:
nit: I think we can add `let copy = publisher.copy() as? PublisherInfo` and get rid of cast force

Reply:
Dont think this is necessary, `PublisherInfo` conforms to `NSCopying` and is auto-generated to return the same type on brave-core side, it would never fail